### ISSUE TITLE
Feature suggestion: skeletonization of output segmentation

### ISSUE THREAD

**Issue by LorenzLamm** [teamtomo/membrain-seg#26]
As discussed in https://github.com/teamtomo/membrain-seg/issues/24 , it might be nice to have an option (or even the default?) to output skeletonized outputs of the membrane segmentations instead of voxel-wise segmentations.

These skeletons would allow for membrane analysis regardless of their thickness, and also allow the user to control the membrane thickness by "growing" them using image dilation.

<img width="1035" alt="skeletonization_snap" src="https://github.com/teamtomo/membrain-seg/assets/34575029/6dd050f2-82f2-4f51-a8ba-2d7816e7c048">

**Comment by alisterburt** [comment]
for my own clarity, the skeletonised output is still a voxel-grid, right? just it has a maximum possible thickness?

**Comment by LorenzLamm** [comment]
Yes, correct. It's still on a voxel grid and should get as thin as possible while still maintaining the connectivity of the original segmentation.

**Comment by rdrighetto** [comment]
I just became aware of this feature request, as Wojciech in our lab just suggested the same thing. After reading the discussion in #24, I think it would be totally awesome to have the skeletonization (and further regrowing to an arbitrary thickness) as an option.
However, I would not make this the default behavior. A skeletonized segmentation is not directly related to the experimental data anymore, so I think the input/output causality becomes more difficult for users to understand. Also, often growing all membranes to the same thickness is not what you want, since varying thickness does happen in real data, can depend on the viewing direction, etc.
Still, I know it can be useful in many scenarios for analysis, and for more standardized performance comparison. So I suggest having this feature as an optional postprocessing operation.
